### ISSUE TITLE
Retry on initial connect

### ISSUE THREAD

**Issue by tylerrinnan-wf** [nats-io/go-nats#310]
The go-nats client attempts to connect to each url in a server pool only once. For those that are running a NATS cluster behind a DNS/ELB, this can mean there is only one network address configured for the cluster. If the connect attempt fails due to a transient network failure/routed to a bad container, the client will not retry the attempt and gives up with an error.

## **Current behavior**:
https://github.com/nats-io/go-nats/blob/master/nats.go#L1051-L1080

## **Requested Behavior**:
If none of the addresses in the `srvPool` slice succeeded in a valid connection, an attempt is retried in the form of iterating over the `srvPool` slice n amount of times. A configuration option for max connect retries could be used here. Alternatively, the retry loop could be limited to only the case where one server is provided to the client.

**Comment by tylertreat** [comment]
I think we've determined this is something we want to do but haven't yet prioritized the work. There is further discussion about this [here](https://github.com/nats-io/go-nats/issues/195). The thought is this could be introduced in a backward compatible way as a new option to the `Connect()` call. We would probably want to determine what that would look like.

**Comment by kozlovic** [comment]
To be consistent with the retry logic, we could introduce a `MaxConnect` option and `MaxConnects()` function to set the option, or depart from that and have `ConnectRetries` for option and function.
Regardless of the naming, as think introducing an option and keep the current behavior is the way to go.

**Comment by ColinSullivan1** [comment]
+1 for introducing a new option; current behavior shouldn't change.  IMO being congruent with the retry option with `MaxConnect` would be cleanest.

**Comment by ColinSullivan1** [comment]
This issue should resolve #195.

**Comment by GlennMatthysECrane** [comment]
#195 and #310 should be merged as they are the same thing.

**Comment by tylertreat** [comment]
I've closed this issue as it is a duplicate of #195. Please follow discussion there.
